core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -112,6 +112,53 @@ class TransactionsTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testReadCommittedConsumerShouldNotSeeUndecidedData(): Unit = {
```

Comment:
Admittedly the terminology can be confusing (especially with the overload of "committed" data).  The term "undecided"refers to data which has yet to be aborted or committed. "Uncommitted" in the "read_uncommitted" sense refers to all data in the log, regardless of its transaction commit status (i.e. whether committed, aborted, or undecided). The LSO is intended to keep the read_committed consumer from seeing undecided data.